server/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -1030,13 +1036,44 @@ private static void ensureAtomicMoveSupported(final NodePath[] nodePaths) throws
         }
     }
 
-    /**
-     * Resolve the custom path for a index's shard.
-     * Uses the {@code IndexMetaData.SETTING_DATA_PATH} setting to determine
-     * the root path for the index.
-     *
-     * @param indexSettings settings for the index
-     */
+    private void ensureNoIndexData(final NodePath[] nodePaths) throws IOException {
+        List<Path> badDataPaths = new ArrayList<>();
```

Comment:
maybe `dataPathsWithShards`?

Reply:
Ended with shardDataPaths, since it contains the path to the shard data.